hack/update-codegen.sh

```diff
@@ -19,11 +19,14 @@ set -o nounset
 set -o pipefail
 
 GO_CMD=${1:-go}
-KUEUE_ROOT=$(realpath $(dirname ${BASH_SOURCE[0]})/..)
-CODEGEN_PKG=$($GO_CMD list -m -f "{{.Dir}}" k8s.io/code-generator)
-
-cd $(dirname ${BASH_SOURCE[0]})/..
-
+KUEUE_ROOT=$(realpath $(dirname "${BASH_SOURCE[0]})"/..))
+if [ -d "vendor" ]; then
```

Comment:
why do you have a vendor folder?

Reply:
> why do you have a vendor folder?


"The 'go mod vendor' command automatically generates the 'vendor' directory. If this command is not used to copy dependencies to the project space, the GoLang editor cannot automatically locate dependencies such as 'sigs.k8s.io/controller-runtime', resulting in red error indicators indicating that the dependencies cannot be found."